### ISSUE TITLE
Errors when obfuscating input with string/hex_hash -> string/file_glob or string/folder_glob

### ISSUE THREAD

**Issue by capnspacehook** [Bashfuscator/Bashfuscator#3]
When using bashfuscator in a command like this:

`bashfuscator -c "echo hi" --choose-mutators string/hex_hash string/file_glob --test --clip`

The test output will sometimes look something like this:

```
/bin/bash: line 42: printf: missing hex digit for \x
/bin/bash: line 51: printf: missing hex digit for \x
/bin/bash: line 56: printf: missing hex digit for \x
/bin/bash: line 62: printf: missing hex digit for \x
/bin/bash: line 117: printf: missing hex digit for \x
/bin/bash: line 213: printf: missing hex digit for \x
/bin/bash: line 266: printf: missing hex digit for \x
/bin/bash: line 0: printf: missing hex digit for \x
/bin/bash: line 0: printf: missing hex digit for \x
/bin/bash: line 0: printf: missing hex digit for \x
/bin/bash: line 0: printf: missing hex digit for \x
/bin/bash: line 0: printf: missing hex digit for \x
/bin/bash: line 0: printf: missing hex digit for \x
/bin/bash: line 0: printf: missing hex digit for \x
hi
```

This stems from the fact that the string/hex_hash Mutator generates `printf` statements like `printf "\x$(...)"`, and the string/file_glob and string/folder_glob Mutators break the input commands into chunks and write them to files using `printf`. 

Therefore, string/file_glob or string/folder_glob will inevitably try to use `printf` a string with `\x` in it, and raise the `missing hex digit for \x` error because no digit precedes the `\x` string.

@cpbarker and I started to find a fix by changing all occurances of `printf --` to `printf '%s'` in the string/file_glob and string/folder_glob Mutators. This helps because the `'%s'` argument makes `printf` interpret the next argument as a literal string, so that `\x` isn't parsed by `printf` as an escape sequence.

**Comment by capnspacehook** [comment]
@343iChurch I *think* I (accidentally) fixed this issue in one of the commits I made on the dev branch today, can you test and verify?

**Comment by 343iChurch** [comment]
@capnspacehook verified this issue is "fixed" as of 6891379, but the fix appears to only work in Debian-based distros. Have tested in Ubuntu, Debian, and Ubuntu WSL. Fedora/RHEL/Centos still have the old issue, may be releated to issue #5.
